plugins/controlboard/src/ControlBoardDriverPidControl.cpp

```diff
@@ -11,20 +11,22 @@
 #include <yarp/os/LogStream.h>
 #include <gazebo/physics/Joint.hh>
 #include <gazebo/transport/Publisher.hh>
-
+namespace {
+    YARP_LOG_COMPONENT(GAZEBOCONTROLBOARD, "gazebo-yarp-plugins.plugins.GazeboYarpControlBoard")
```

Comment:
This `YARP_LOG_COMPONENT` macro is duplicated among several different .cpp files, is that normal/expected?

Reply:
Actually it was a mistake. But there are 2 options:
- if we want to log also the file we can keep them there and add the class to component
- otherwise I can put it in ControlBoard.hh file

Reply:
I think keeping a mapping between a class and a component is ok, no need for further granularity. However, from what I understand `YARP_LOG_COMPONENT` defines  a function with a static member, so you can't put it in a header, as this will result in ODR (one definition rule)-violations.

Reply:
Actually I tried to put it inside the h file and it gaves me some undeclared error, so I think that sould be kept inside every file

Reply:
If you see https://github.com/robotology/yarp/blob/ea6ed180ddb102679d8b8492fc40329719d7b8e4/src/libYARP_os/src/yarp/os/LogComponent.h#L78, you see that `YARP_LOG_COMPONENT` **defines** a function, while `YARP_DECLARE_LOG_COMPONENT` declares it. So I think you should do the same thing that you do if you are creating a function: you don't need to copy&paste it in every file, but rather declare it in the header, and then placing the implementaton in only a single cpp file.